modules/core/keeper/msg_server.go

```diff
@@ -488,6 +488,11 @@ func (k *Keeper) RecvPacket(goCtx context.Context, msg *channeltypes.MsgRecvPack
 		return nil, errorsmod.Wrap(err, "receive packet verification failed")
 	}
 
+	// performance: return early for the redundant relayer ante handler
+	if ctx.IsCheckTx() || ctx.IsReCheckTx() {
+		return &channeltypes.MsgRecvPacketResponse{Result: channeltypes.SUCCESS}, nil
+	}
```

Comment:
if its redundant we will already have returned in switch above on L485 and if its not considered successful for mempool insertion 👍🏻